packages/atomic/src/components/commerce/atomic-commerce-interface/atomic-commerce-interface.tsx

```diff
@@ -0,0 +1,480 @@
+import {
+  LogLevel,
+  Search,
+  Unsubscribe,
+  UrlManager,
+  buildSearch,
+  buildSearchUrlManager,
+  getOrganizationEndpoints as getOrganizationEndpointsHeadless,
+  updateQuery,
+  CommerceEngine,
+  CommerceEngineConfiguration,
+  buildCommerceEngine,
+} from '@coveo/headless/commerce';
+import {
+  Component,
+  Prop,
+  h,
+  Listen,
+  Method,
+  Watch,
+  Element,
+  State,
+  setNonce,
+} from '@stencil/core';
+import i18next, {i18n} from 'i18next';
+import {InitializeEvent} from '../../../utils/initialization-utils';
+import {
+  SafeStorage,
+  StandaloneSearchBoxData,
+  StorageItems,
+} from '../../../utils/local-storage-utils';
+import {ArrayProp} from '../../../utils/props-utils';
+import {CommonBindings, NonceBindings} from '../../common/interface/bindings';
+import {i18nCompatibilityVersion} from '../../common/interface/i18n';
+import {
+  BaseAtomicInterface,
+  CommonAtomicInterfaceHelper,
+} from '../../common/interface/interface-common';
+import {getAnalyticsConfig} from './analytics-config';
+import {AtomicCommerceStore, createAtomicCommerceStore} from './store';
+
+const FirstSearchExecutedFlag = 'firstSearchExecuted';
+
+export type CommerceInitializationOptions = CommerceEngineConfiguration;
+export type CommerceBindings = CommonBindings<
+  CommerceEngine,
+  AtomicCommerceStore,
+  HTMLAtomicCommerceInterfaceElement
+> &
+  NonceBindings;
+
+/**
+ * The `atomic-commerce-interface` component is the parent to all other atomic commerce components in a commerce page. It handles the headless search engine and localization configurations.
+ */
+@Component({
+  tag: 'atomic-commerce-interface',
+  styleUrl: 'atomic-commerce-interface.pcss',
+  shadow: true,
+  assetsDirs: ['lang'],
+})
+export class AtomicCommerceInterface
+  implements BaseAtomicInterface<CommerceEngine>
+{
+  private urlManager!: UrlManager;
+  private searchStatus!: Search;
+  private unsubscribeUrlManager: Unsubscribe = () => {};
+  private unsubscribeSearchStatus: Unsubscribe = () => {};
+  private initialized = false;
+  private store = createAtomicCommerceStore();
+  private commonInterfaceHelper: CommonAtomicInterfaceHelper<CommerceEngine>;
+
+  @Element() public host!: HTMLAtomicCommerceInterfaceElement;
+
+  @State() public error?: Error;
+
+  /**
+   * The type of the interface.
+   * - 'search': Indicates that the interface is used for Search.
+   * - 'product-listing': Indicates that the interface is used for Product listing.
+   */
+  @Prop({reflect: true, mutable: true}) public type:
+    | 'search'
+    | 'product-listing' = 'search';
+
+  /**
+   * A list of non-default fields to include in the query results.
+   *
+   * Specify the property as an array using a JSON string representation:
+   * ```html
+   * <atomic-commerce-interface fields-to-include='["fieldA", "fieldB"]'></atomic-commerce-interface>
+   * ```
+   */
+  @ArrayProp()
+  @Prop({mutable: true})
+  public fieldsToInclude: string[] | string = '[]';
```

Comment:
Fields to include are not necessary I think. They are configured in the commerce service: Front end code does not need to be configured to request them.

To confirm with routemaster/commerce service documentation